src/frontend/src/optimizer/plan_node/batch_sort_agg.rs

```diff
@@ -65,6 +61,12 @@ impl BatchSortAgg {
 
         assert_eq!(input_order.field_order.len(), logical.group_key().len());
 
+        let order = logical
+            .i2o_col_mapping()
+            .rewrite_provided_order(&input_order);
```

Comment:
@st1page Could you confirm that my assumption here is correct, that SortAgg will not change the ordering of its input. So we just remap i2o columns and the order remains the same?

Reply:
Yes